docs/production/authentication-methods.md

```diff
@@ -698,11 +700,28 @@ another IdP.
    /home/zulip/deployments/current/manage.py logout_all_users
    ```
 
+#### SP-initiated Single Logout
+
+After configuring IdP-initiated Logout, you only need to set
+`SAML_ENABLE_SP_INITIATED_SINGLE_LOGOUT = True` in
+`/etc/zulip/settings.py` to also enable SP-initiated Logout. When this
+is active, a user who logged in to Zulip via SAML, upon clicking
+"Logout" in the Zulip web app will be redirected to the IdP's Single
+Logout endpoint with a `LogoutRequest`. If a successful
+`LogoutResponse` is received back, their current Zulip session will be
+terminated.
+
+Note that this doesn't work in the case of desktop and mobile application
+and is reserved to the browser.
+
 #### Caveats
 
-- This beta doesn't support using `SessionIndex` to limit which
-  sessions are affected; it always terminates all logged-in sessions
-  for the user identified in the `NameID`.
+- This implementation doesn't support using `SessionIndex` to limit which
+  sessions are affected; in IdP-initiated Logout it always terminates
+  all logged-in sessions for the user identified in the `NameID`.
+  In SP-initiated Logout this simply means that Zulip does not include
+  `SessionIndex` in the `LogoutRequest` to the IdP - however, that doesn't
+  seem to cause any undesired behavior with Keycloak.
```

Comment:
Should we open an issue for `SessionIndex` support? I'm not entirely sure what it would do, and having that issue to refer to might help me improve this documentation.

Reply:
`SessionIndex` is actually implemented (and that's what addresses the security issue that blocked this from merging in 6.0) in `saml: Save SessionIndex in session and use when making a LogoutRequest. `. The commit message explains a bit what SessionIndex is for:
```quote
This is a useful improvement in general for making correct
LogoutRequests to Idps and a necessary one to make SP-initiated logout
fully work properly in the desktop application. During desktop auth
flow, the user goes through the browser, where they log in through their
IdP. This gives them a logged in  browser session at the IdP. However,
SAML SP-initiated logout is fully conducted within the desktop
application. This means that proper information needs to be given to the
the IdP in the LogoutRequest to let it associate the LogoutRequest with
that logged in session that was established in the browser. SessionIndex
is exactly the tool for that in the SAML spec
```

The thing with reviewing this PR is that it was built very iteratively, so it adds documentation in some commits that then gets changed in the next commits that implement further capabilities. I could squash all the commits, but then it becomes a pretty huge and complicated implementation in one chunk, not sure if that would be easier to review? :confounded: For reviewing the documentation itself, it might be best  to look at it in the `Files changed` tab rather than per-commit to just see the final state.

Reply:
Ahh, thanks for clarifying that; I reviewed it viewing the "Files changed" part and found that easier to follow. I think it's probably fine with the structure as written, now that I understand it.